app/src/main/java/com/markkolenbrander/capstonenewsapp/ArticleDataManager.kt

```diff
@@ -0,0 +1,32 @@
+package com.markkolenbrander.capstonenewsapp
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.preference.PreferenceManager
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.markkolenbrander.capstonenewsapp.models.Article
+
+class ArticleDataManager(application: Application) : AndroidViewModel(application){
+
+    private val context = application.applicationContext
+
+    private val sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
```

Comment:
To prevent the context from being leaked you can actually just pass in `context` here in to the constructor and reference it in the PreferenceManager method call.  This way you're not holding on to a reference of `context` at all.  You'll use it once to get the sharedPrefs and release it.

```suggestion
class ArticleDataManager(context: Context) : AndroidViewModel(application){

    private val sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context)
```